src/core/control/ExportHelper.cpp

```diff
@@ -96,30 +86,28 @@ auto exportPdf(Document* doc, const char* output, const char* range, const char*
 
     GFile* file = g_file_new_for_commandline_arg(output);
 
-    std::unique_ptr<XojPdfExport> pdfe = XojPdfExportFactory::createExport(doc, nullptr);
-    pdfe->setExportBackground(exportBackground);
     auto path = fs::u8path(g_file_peek_path(file));
     g_object_unref(file);
 
-    bool exportSuccess = 0;  // Return of the export job
-
-    pdfe->setLayerRange(layerRange);
-
+    PageRangeVector exportRange = ElementRange::parse("1-" + std::to_string(doc->getPageCount()), doc->getPageCount());
```

Comment:
This parsing is unnecessary. Simply emplace an ElementRange with the given bounds.
Also, this should probably go in an `else` block after the following `if` statement.